NGitLab/IProtectedBranchesClient.cs

```diff
@@ -0,0 +1,15 @@
+﻿using NGitLab.Models;
+
+namespace NGitLab
+{
+    public interface IProtectedBranchesClient
```

Comment:
Please rename to `IProtectedBranchClient`. As per most other client interfaces, it should be named with a singular noun followed by `Client`.

Rename derived types and methods too accordingly.